app.ts

```diff
@@ -0,0 +1,265 @@
+			if (inputValue !== '') {
+				await searchRecordsAndResize(recordCount);
+			}
+		}, 250);
+	});
+}
+
+/** display records that fit the screen */
+async function displayRecords(recordCount: number): Promise<void> {
+	$('#loader').show();
+	const inputValue = $('#searchInput').val() as number;
+	const calculatedRows = calculatingRows();
+	const { firstNumber, lastNumber } = calculateFirstAndLastNumbers(calculatedRows, recordCount);
+	updateArrowVisibility(firstNumber, lastNumber, recordCount);
+	const records = await fetchAndDisplayRecords(firstNumber, lastNumber, inputValue);
+	$('#page').empty().append(`Showing record: ${firstNumber} - ${lastNumber}`);
+	$('#loader').hide();
+}
+
+function calculateFirstAndLastNumbers(calculatedRows: number, recordCount: number) {
+	let firstNumber, lastNumber;
+	 if (myClass.firstNumber < 0 || myClass.firstNumber > recordCount) {
+		firstNumber = 0;
+	} else {
+		firstNumber = myClass.firstNumber;
+	}
+	lastNumber = firstNumber + calculatedRows - 1;
+	if (lastNumber >= recordCount) {
+		lastNumber = recordCount;
+		firstNumber = lastNumber - (calculatedRows - 1)
+	}
+	return { firstNumber, lastNumber };
+}
+
+function updateArrowVisibility(firstNumber: number, lastNumber: number, recordCount: number) {
+	if (firstNumber === 0) {
+		$('.arrow-left').hide();
+	} else {
+		$('.arrow-left').show();
+	}
+
+	if (lastNumber >= recordCount) {
+		$('.arrow-right').hide();
+	} else {
+		$('.arrow-right').show();
+	}
+}
+
+async function fetchAndDisplayRecords(firstNumber: number, lastNumber: number, inputValue: number) {
+	const records = await myClass.fetchRecords(firstNumber, lastNumber);
+	const tbody = $("tbody");
+	tbody.empty();
+	for (const record of records) {
+		/** creates row for each record*/ 
+		$("tbody").append(`<tr class="row"></tr>`);
+		const lastRow = $(".row:last");
+		for (const value of record) {
+			/** assign each record to their column in a specified row */
+			lastRow.append(`<td>${value}</td>`);
+		}
+		if (record.includes(inputValue)) {
+			/** highlights the searched row */
+			lastRow.css('background-color', '#DDC0B4');
+		}
+		$("tbody").append(lastRow);
+	}
+	return records;
+}
+
+/** recalculates the record range that includes inputValue */
+async function searchRecordsAndResize(recordCount: number): Promise<void> {
+	let inputValue = $('#searchInput').val() as number;
+	if (inputValue < 0 || inputValue > recordCount) {
+		$('.modal').css('display', 'block');
+		$('.content').append(`<p>${inputValue} is not a number within the range. Please try a different number</p>`);
+		$('#page').empty().append(`Showing record: ${myClass.firstNumber} - ${myClass.lastNumber}`);
+		$('#searchInput').val('');
+	}
+	let calculatedRows = calculatingRows();
+	/** divides the calculated max rows in half*/
+	const halfRange = Math.floor(calculatedRows / 2);
+	myClass.firstNumber = Math.max(0, inputValue - halfRange);
+	myClass.lastNumber = Math.min(recordCount, myClass.firstNumber + (calculatedRows - 1));
```

Comment:
If I understand correctly, you show the searched record in the middle of the grid and highlight it. In IMQS we show the most relevant record we're searching for at the top of the grid without formatting. But you can still keep the formatting because it's cool.

Reply:
This I remember from the advice that was given to me for my own onboarding project, weird.